safemode/safemode.go

```diff
@@ -0,0 +1,255 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package safemode
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	chaostypes "github.com/DataDog/chaos-controller/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Safemode interface {
+	// CreationSafetyNets will look at all the specific safety nets corresponding to the creation of a new disruption
+	CreationSafetyNets() ([]string, error)
+	// GetTypeSpec will grab the necessary spec depending on the type (e.g. SafemodeNetwork will grab NetworkDisruptionSpec)
+	// and grab the disruption itself for data such as the kubernetes namespace the disruption is running on
+	GetTypeSpec(disruption v1beta1.Disruption)
+	// GetKubeClient grab the kube client for functions that require state information from k8s system
+	GetKubeClient(client client.Client)
```

Comment:
These two functions appear to be setters, not getters? I think you're fine just putting `Init` in the interface in that case

Reply:
Good Point, adding their functionality to init and removing the setters.